tests/test_kitti_format.py

```diff
@@ -459,21 +454,62 @@ def __iter__(self):
 
     @mark_requirement(Requirements.DATUM_GENERAL_REQ)
     def test_can_save_and_load_with_no_save_images_detection(self):
-        class TestExtractor(TestExtractorBase):
-            def __iter__(self):
-                return iter([
-                    DatasetItem(id='b', subset='val', image=np.ones((5, 5, 3)),
-                        annotations=[
-                            Bbox(0, 0, 3, 3, label=2, attributes={
-                                'truncated': True, 'occluded': False
-                            })
-                        ])
+        source_dataset = Dataset.from_iterable([
+            DatasetItem(id='b', subset='val', image=np.ones((5, 5, 3)),
+                annotations=[
+                    Bbox(0, 0, 3, 3, label=0, attributes={
+                        'truncated': True, 'occluded': False
+                    })
                 ])
-
-            def categories(self):
-                return make_kitti_detection_categories()
+        ], categories=['label_0'])
 
         with TestDir() as test_dir:
-            self._test_save_and_load(TestExtractor(),
+            self._test_save_and_load(source_dataset,
                 partial(KittiConverter.convert, tasks=KittiTask.detection,
                     save_images=False), test_dir)
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_can_save_and_load_segmentation_with_unordered_labels(self):
+        source_label_map = {
+            'background': (0, 0, 0),
+            'label_1': (10, 10, 10),
+            'label_0': (20, 20, 20)
+        }
+
+        source_dataset = Dataset.from_iterable([
+            DatasetItem(id='a', image=np.ones((1, 5, 3)),
+                annotations=[
+                    Mask(image=np.array([[1, 0, 0, 0, 0]]),
+                        attributes={'is_crowd': False}, label=0, id=1),
+                    Mask(image=np.array([[0, 1, 0, 0, 0]]),
+                        attributes={'is_crowd': False}, label=1, id=1),
+                    Mask(image=np.array([[0, 0, 1, 1, 1]]),
+                        attributes={'is_crowd': False}, label=2, id=2)
+                ]
+            )
+        ], categories=make_kitti_categories(source_label_map))
+
+        expected_label_map = {
+            'background': (0, 0, 0),
+            'label_0': (20, 20, 20),
+            'label_1': (10, 10, 10)
+        }
```

Comment:
Why the labels are sorted here?

Reply:
Because in this solution label indexes may change after the import-export procedure. I agree that this is a disadvantage of such a solution. But I think this is better than requiring the user to fill in an additional file with a list of tags.